### PR TITLE
mm/kasan: replace load/store methods to macro 

### PR DESCRIPTION
**Pull request by anchao**
## Summary

mm/kasan: replace load/store methods to marco 

## Impact

N/A

## Testing

ci-check

### CODE REVIEW

**Review by pkarashchenko — APPROVED** [review]

File: mm/kasan/kasan.c [hunk 1]
```diff
@@ -376,52 +276,34 @@ void __asan_storeN(FAR void *addr, size_t size)
...
+  void __asan_load##size(FAR void *addr) \
+  { \
+    __asan_load##size##_noabort(addr); \
+  } \
+  void __asan_store##size(FAR void *addr) \
+  { \
+    __asan_store##size##_noabort(addr); \
+  }
+
+DEFINE_ASAN_LOAD_STORE(1);
```
Comment (xiaoxiang781216) [inline mm/kasan/kasan.c:305]: remove the end ;

**Comment by acassis** [comment]
Guys, although this is a good example where macro can be used to create code in a smarter way, I don't know if this is a good move. Have you seen "how NASA writes space-proof code": https://www.youtube.com/watch?v=GWYhtksrmhE

They have basically 10 rules (called "The Power of Ten") and one of those rules is to avoid using complex macro that makes code difficult to use with static analyze software.

No strong opinion about it, since we don't follow most of those 10 rules, but just to raise our awareness about it.

**Comment by anchao** [comment]
> No strong opinion about it, since we don't follow most of those 10 rules, but just to raise our awareness about it.

@acassis Thanks for your suggestion. Actually this is the practice in the linux kernel. The macro expansion of kasan just removes the relevant redundant functions, which will be more clear for developers read the code.

https://github.com/torvalds/linux/blob/master/mm/kasan/generic.c#L238-L260

File: mm/kasan/kasan.c (hunk 1 above)
Comment (anchao) [reply mm/kasan/kasan.c]: Done

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**